### PR TITLE
build.yml: Use a newer build toolchain

### PR DESCRIPTION
**Pull request by jepler**
.. the new version is not offered in .deb form, so the installation process also changes

Problems:
 * several m0 builds (pirkey, trinket, pyruler) no longer fit
 * this uses the tarball direct from arm; I think we'd prefer to have it in storage managed by circuitpython project, like the current toolchain .deb

### CODE REVIEW

**Comment by jepler** [comment]
@tannewt let me know what direction you'd like to see this go in, now that we know there are failures due to binary size increases

**Comment by tannewt** [comment]
Let's close this for now. We can update along with CP 6. The Pirkey is hundreds of bytes over which will be hard to recover.

**Comment by jepler** [comment]
@tannewt makes sense, couldn't have been predicted without the CI run...

**Comment by jepler** [comment]
I'm looking at this again, there are some optimization flags we can set so that gcc9 behaves closer to gcc7 in terms of code size.  I think this may let the most constrained m0 boards fit again.

File: ports/atmel-samd/boards/pewpew10/mpconfigboard.mk [hunk 1]
```diff
@@ -23,3 +23,5 @@ CIRCUITPY_USB_MIDI = 0
 SUPEROPT_GC = 0
 
 FROZEN_MPY_DIRS += $(TOP)/frozen/pew-pewpew-standalone-10.x
+
+CFLAGS_BOARD = --param inline-unit-growth=15 --param max-inline-insns-auto=15
```
Comment (tannewt) [inline ports/atmel-samd/boards/pewpew10/mpconfigboard.mk]: How about factoring these out for all small boards?

File: ports/atmel-samd/nomakefile [hunk 2]
```diff
@@ -0,0 +1,30 @@
+MAKEFLAGS += -j5
+BOARD ?= metro_m4_express
+BOARD ?= feather_m0_express
+DEBUG = 0
+#$(info I: Building for $(BOARD) $(DEBUG))
+include Makefile
```
Comment (tannewt) [inline ports/atmel-samd/nomakefile]: What is this file for?

Comment (jepler) [reply ports/atmel-samd/nomakefile]: Mistaken addition, I'll rebase it out

File: ports/atmel-samd/boards/pewpew10/mpconfigboard.mk (hunk 1 above)
Comment (jepler) [reply ports/atmel-samd/boards/pewpew10/mpconfigboard.mk]: I ended up with 3 different sets of flags for 5 boards.  My gut said this wasn't worth worrying about DRY.

+CFLAGS_BOARD = --param inline-unit-growth=15 --param max-inline-insns-auto=15
+CFLAGS_BOARD = --param inline-unit-growth=15 --param max-inline-insns-auto=12
+CFLAGS_BOARD = --param inline-unit-growth=15 --param max-inline-insns-auto=15
+CFLAGS_BOARD = --param inline-unit-growth=15 --param max-inline-insns-auto=20
+CFLAGS_BOARD = --param inline-unit-growth=15 --param max-inline-insns-auto=20

**Comment by jepler** [comment]
Sigh another board that doesn't fit

File: ports/atmel-samd/boards/pewpew10/mpconfigboard.mk (hunk 1 above)
Comment (tannewt) [reply ports/atmel-samd/boards/pewpew10/mpconfigboard.mk]: Seems like the first param can be factored out. Adding new small builds will be easier if these CFLAGS automatically apply.

**Pull request merged**